### ISSUE TITLE
SC2142: Ignore calls to awk

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1238]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2142
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
Kind of: https://github.com/koalaman/shellcheck/issues/728

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
alias hu=" history | awk '{ \$1 = \"\" ; print }' | awk '!a[\$0]++' | cut -c 2-"
```

#### Here's what shellcheck currently says:
```sh
$ shellcheck myscript
 
Line 2:
alias hu=" history | awk '{ \$1 = \"\" ; print }' | awk '!a[\$0]++' | cut -c 2-"
      ^-- SC2142: Aliases can't use positional parameters. Use a function.

$ 
```

#### Here's what I wanted or expected to see:

Nothing

As obvious, I am trying to extract history, minus duplicates, in-order. I am not sure if I can write `awk '{ $1 = "" ; print }'`, since I need to open-up with a quote. I could of course do `'"'"'`, but I think that would be ugly.

**Comment by koalaman** [comment]
You're right, of course.

ShellCheck's design decision is to generally avoid introspection of even trivial forms of indirect evaluation, to avoid giving the impression that this should work more generally. In this case, it contradicts that enough to look for `$1` in aliases, but not enough to actually find the context, thereby resulting in an annoying false positive.

Here's the wiki's suggestion:

>If your alias ends up quoting the value, e.g. `alias cut_first="awk '{print \$1}'"`, you can technically ignore this error. However, you should consider turning this alias into a more readable function instead: 
`cut_first() { awk '{print $1}' "$@"; }`

I'm not sure if it would be better to blacklist any alias that contains the word `awk`, try to parse `alias` commands as stand-alone function bodies, try to parse the alias in the context where it's used (like a shell), or just suggest functions for any use of aliases. They all have their cons.

It's infrequent enough that I'll leave it for another time. Meanwhile, consider a function. Unlike the wiki's dubious example, it actually *would* be more readable :P

**Comment by sudoshindo** [comment]
"labelled as fix later"

Can we fix it now?
I ran into nearly the exact same situation. I'm also trying to write some shell history related alias.
